src/components/DetailsOneRow.svelte

```diff
@@ -25,16 +29,21 @@
   <td>{addressLine.rank_address}</td>
   <td>{formatAdminLevel(addressLine.admin_level)}</td>
   <td>{@html formatDistance(addressLine.distance, bDistanceInMeters)}</td>
-  <td>{#if addressLine.osm_id}<DetailsLink feature={addressLine}>details</DetailsLink>{/if}</td>
+  <td>
+    {#if addressLine.osm_id}
+      <DetailsLink feature={addressLine}>details</DetailsLink>
+    {:else if addressLine.type.match(/^country/)}
+      <a href={search_by_name_url('country', addressLine.localname)}>search by name</a>
+    {:else if addressLine.type === 'postcode'}
+      <a href={search_by_name_url('postalcode', addressLine.localname)}>search by name</a>
```

Comment:
Add a url_params parameter to PageLink and use that? Gets you a proper internal link.

Reply:
Added a `params_hash` now